Lib/test/test_importlib/identity.py

```diff
@@ -0,0 +1,521 @@
+    as well as into a more succinct structured format - two lists of
+    (name, email) tuples, one for authors and the other for maintainers.
+
+    If ``simple_chars``, restrict the list of allowed characters to
+    reduce the Hypothesis search space and permit easier visualization.
+    If ``debug_entries``, return the underlying lists of ``Entry``
+    instances in addition to the unstructured string and structured list
+    representations.
+
+    This strategy will generate the most permissible identities possible:
+    unbalanced quote characters, unescaped special characters, omitted
+    identity information, empty identities, and so forth.
+    """
+    text = ""
+    authors = set()
+    maintainers = set()
+    entry_dict = dict()
+    state = types.SimpleNamespace(simple_chars=simple_chars)
+
+    combinations = zip(
+        ((i, j) for i in (name_entry, ident_entry) for j in (authors, maintainers)),
+        ("Author", "Maintainer", "Author-email", "Maintainer-email"),
+    )
+
+    for (entry_type, entry_target), entry_prefix in combinations:
+        entries, data_set, data_str = draw(identity_entries(state, entry_type))
+        entry_dict[entry_prefix] = entries
+        if not entries:
+            continue
+        text += entry_prefix + ": " + data_str + "\n"
+        entry_target |= data_set
+
+    result = (text, authors, maintainers)
+    if debug_entries:
+        result = (entry_dict, *result)
+    return result
+
+
+@st.composite
+def identity_entries(draw, state, entry_strategy):
+    """
+    Generate a list of processed identity entries using
+    ``entry_strategy``.
+    """
+    sample = draw(st.lists(entry_strategy(state), min_size=0, max_size=10))
+    for entry in sample[:-1]:
+        entry.append(Token(identity_entries, ", ", True))
+    unbalance(sample)
+    lstrip(sample)
+    return process(sample)
+
+
+def process(entries):
+    """
+    Process a list of entries into structured and unstructured
+    formats. Filter empty (``None``) entries from the structured
+    format. Convert token categories to strings for simpler
+    visualization.
+    """
+    text = ""
+    data = set()
+
+    for entry in entries:
+        entry_text = entry.as_text
+        entry_data = entry.as_data
+        for token in entry:
+            token.category = token.category.__name__
+        text += entry_text
+        if all(elem is None for elem in entry_data):
+            continue
+        data.add(entry_data)
+
+    return (entries, data, text)
+
+
+def lstrip(entries):
+    """
+    The PackageMetadata implementation removes leading whitespace
+    from the first entry, so we do the same. This should not alter
+    the parsing of the entry.
+    """
+    entry = entries[0] if entries else []
+    strip = False
+    for token in entry:
+        value = token.value.lstrip()
+        strip = strip or token.value != value
+        token.value = value
+        if token.value:
+            break
+    return strip
```

Comment:
```suggestion
    for token in entry:
        token.value = token.value.lstrip()
        if token.value:
            break
```
The `strip` bool was never used, so this can be more compact (although I think using `:=` would be too far).